### PR TITLE
Set ready status when a replica count is zero

### PR DESCRIPTION
**Pull request by kylape**
When a specified replica count is zero, we should mark the deployment as
ready.

I've also taken the liberty to add some comments to the
DeploymentStatusChecker function because the different checks were
difficult to understand at first (and maybe second) glance.

### CODE REVIEW

File: controllers/cloud.redhat.com/utils/utils.go [hunk 1]
```diff
@@ -254,22 +254,43 @@ func MakePVC(pvc *core.PersistentVolumeClaim, nn types.NamespacedName, labels ma
...
+	}
+
+	if deployment.Spec.Replicas != nil {
+		if *deployment.Spec.Replicas == 0 {
+			// Since there's no replicas, there's nothing to do to get ready
+			return true
 		}
-		if deployment.Status.AvailableReplicas < deployment.Status.UpdatedReplicas {
+
+		if deployment.Status.UpdatedReplicas < *deployment.Spec.Replicas {
```
Comment (psav) [inline controllers/cloud.redhat.com/utils/utils.go:271]: I'm guessing if `Replicas` is 0, then `UpdatedReplicas` will never be set at all?

File: controllers/cloud.redhat.com/utils/utils.go [hunk 2]
```diff
@@ -254,22 +254,43 @@ func MakePVC(pvc *core.PersistentVolumeClaim, nn types.NamespacedName, labels ma
 	}
 }
 
-// DeploymentStatusChecker takes a deployment and returns True if it is deemed ready by the logic
-// in the function.
+// DeploymentStatusChecker takes a deployment and returns True if it is deemed
+// ready by the logic in the function.
```
Comment (psav) [inline controllers/cloud.redhat.com/utils/utils.go:258]: Shortening the line length? I thought we used 100?

Comment (kylape) [reply controllers/cloud.redhat.com/utils/utils.go:258]: I can change it to 100

File: controllers/cloud.redhat.com/utils/utils.go (hunk 1 above)
Comment (kylape) [reply controllers/cloud.redhat.com/utils/utils.go:271]: I haven't tested it, but if there are zero replicas specified, I wouldn't expect there to be any replicas to update.

**Review by BlakeHolifield — APPROVED** [review]

**Pull request merged**